packages/addon/src/addon.ts

```
@@ -645,69 +690,223 @@ builder.defineStreamHandler(
-        // Compare MB files
-        if (aSize.includes('MB') && bSize.includes('MB')) {
-          const aMB = parseFloat(aSize.match(/[\d.]+/)?.[0] || '0');
-          const bMB = parseFloat(bSize.match(/[\d.]+/)?.[0] || '0');
-          if (aMB > bMB) return -1;
-          if (aMB < bMB) return 1;
-        }
+          // Apply sorting based on user preference
+          switch (sortingPreference) {
+            case 'size_first':
+              // Size first, then quality, then language
+              const sizeCompare = compareSize();
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)